### PR TITLE
Add documentation on the current breaking changes policy.

### PR DESCRIPTION
**Pull request by wiwei**
As a team we've had a policy around breaking changes, though we haven't really written this down until now. This is a summarization of the current state of our stance on breaking changes (tl;dr it's impossible to avoid in all cases, in many cases it's possible to avoid, in the cases where it's not possible to avoid we have to see if the value the change bring is greater than the pain of adoption)

### CODE REVIEW

File: Documentation/Contributing/BreakingChanges.md [hunk 1]
```diff
@@ -1,2 +1,129 @@
...
+- The change in the order of events fired by a class.
+- The rename of any private SerializedField (without a corresponding FormerlySerializedAs tag) or public
+  property on a ScriptableObject (especially changes to profiles).
+- Changing the type of a field on a ScriptableObject (especially changes to profiles).
+- Updates to the namespace or asmdefs of any class or interface.
+- Removal of any prefab or removal of a script on the top level object of a prefab.
+
+### List B
+
+- The asset in question is in the Foundation package (i.e. it's in one of the following folders):
```
Comment (thalbern) [inline Documentation/Contributing/BreakingChanges.md]:
```suggestion
- The asset in question is in the foundation package (i.e. it's in one of the following folders):
```

File: Documentation/Contributing/BreakingChanges.md [hunk 2]
```diff
@@ -1,2 +1,129 @@
...
+
+  - MixedRealityToolkit/
+  - MixedRealityToolkit.Providers/
+  - MixedRealityToolkit.Services/ 
+  - MixedRealityToolkit.SDK/
+
+- The asset in question does not belong to the Experimental namespace.
+
+> Any asset that sits in the Examples package (i.e. part of the MixedRealityToolkit.Examples/ folder)
```
Comment (thalbern) [inline Documentation/Contributing/BreakingChanges.md]:
```suggestion
> Any asset that sits in the examples package (i.e. part of the MixedRealityToolkit.Examples/ folder)
```

File: Documentation/Contributing/BreakingChanges.md [hunk 3]
```diff
@@ -1,2 +1,129 @@
...
+### List B
+
+- The asset in question is in the Foundation package (i.e. it's in one of the following folders):
+
+  - MixedRealityToolkit/
+  - MixedRealityToolkit.Providers/
+  - MixedRealityToolkit.Services/ 
+  - MixedRealityToolkit.SDK/
+
+- The asset in question does not belong to the Experimental namespace.
```
Comment (thalbern) [inline Documentation/Contributing/BreakingChanges.md]:
```suggestion
- The asset in question does not belong to the experimental namespace.
```

File: Documentation/Contributing/BreakingChanges.md [hunk 4]
```diff
@@ -1,2 +1,129 @@
...
+> Any asset that sits in the Examples package (i.e. part of the MixedRealityToolkit.Examples/ folder)
+> is subject to change at any time, as assets there are designed to be copied and viewed by consumers
+> as 'reference implementations' but are not part of the core set of APIs and assets.
+> Assets in the experimental namespace (or more generally, features labelled as experimental) are ones
+> that get published before all due diligence has been done (i.e. tests, UX iteration, documentation) and
+> is published early to get feedback sooner.  However, because they don't have tests and documentation, and
+> because we likely haven't nailed down all of the interactions and designs, we publish them in a state
+> where the public should assume that they can and will change (i.e. be modified, completely removed, etc).
+> 
+> See [Experimental Features](../ExperimentalFeatures.md) for more information.
```
Comment (thalbern) [inline Documentation/Contributing/BreakingChanges.md]:
```suggestion
> See [Experimental features](../ExperimentalFeatures.md) for more information.
```

File: Documentation/Contributing/BreakingChanges.md [hunk 5]
```diff
@@ -1,2 +1,129 @@
...
+
+Note that surface area for breaking changes is very large, it's important to note that having an absolute
```
Comment (thalbern) [inline Documentation/Contributing/BreakingChanges.md]:
```suggestion
As the surface area for breaking changes is very large, it's important to note that having an absolute
```
(there's two times note in that sentence, so i would get rid of the first one)

Comment (wiwei) [reply Documentation/Contributing/BreakingChanges.md]: yay

**Pull request merged**